keel-docker/src/main/kotlin/com/netflix/spinnaker/keel/docker/ContainerProvider.kt

```diff
@@ -21,25 +21,29 @@ import com.fasterxml.jackson.databind.JsonDeserializer
 import com.fasterxml.jackson.databind.annotation.JsonDeserialize
 import com.netflix.spinnaker.keel.api.TagVersionStrategy
 
-@JsonDeserialize(using = ContainerDeserializer::class)
-sealed class Container(
-  open val organization: String,
-  open val image: String
-) {
-  fun repository() = "$organization/$image"
-}
+@JsonDeserialize(using = ContainerProviderDeserializer::class)
+sealed class ContainerProvider
+
+@JsonDeserialize(using = JsonDeserializer.None::class)
+data class ReferenceProvider(
+  val reference: String
+) : ContainerProvider()
 
 @JsonDeserialize(using = JsonDeserializer.None::class)
-data class ContainerWithDigest(
-  override val organization: String,
-  override val image: String,
+data class DigestProvider(
+  val organization: String, // todo eb: should this be name = org/image instead, for consistency?
+  val image: String,
   val digest: String
-) : Container(organization, image)
+) : ContainerProvider() {
+  fun repository() = "$organization/$image"
+}
 
 @JsonDeserialize(using = JsonDeserializer.None::class)
-data class ContainerWithVersionedTag(
-  override val organization: String,
-  override val image: String,
+data class VersionedTagProvider(
```

Comment:
A backwards compatible container provider so that existing delivery configs don't break